packages/@n8n/nodes-langchain/nodes/vendors/GoogleGemini/methods/listSearch.ts

```diff
@@ -64,11 +64,22 @@ export async function imageGenerationModelSearch(
 	this: ILoadOptionsFunctions,
 	filter?: string,
 ): Promise<INodeListSearchResult> {
-	return await baseModelSearch.call(
+	const results = await baseModelSearch.call(
 		this,
-		(model) => model.includes('imagen') || model.includes('image-generation'),
+		(model) =>
+			model.includes('imagen') ||
+			model.includes('image-generation') ||
+			(model.includes('gemini') && model.includes('image')),
 		filter,
 	);
+
+	return {
+		results: results.results.map((r) =>
+			r.name.includes('gemini-2.5-flash-image')
+				? { name: `${r.name} (Nano Banana)`, value: r.value }
```

Comment:
Do we need to include nano banana? We should ideally keep model names to what the API calls them to reduce any possibly confusion if a user tries to use the id or expression option.

Reply:
It was really just because of how this feature is being hyped - all press / marketing I've seen is calling it Nano Banana and obscuring the actual model name - do you think the potential issue would outweigh the discovery benefits?